packages/typespec-python/src/emitter.ts

```diff
@@ -249,6 +261,18 @@ export function getType(context: SdkContext, type: EmitterType): any {
             }
             // need to do discriminator outside `emitModel` to avoid infinite recursion
             handleDiscriminator(context, type, newValue);
+
+            // special case for additionalProperties
+            try {
+                if (newValue.type === "model" && newValue.parents.length > 0 && newValue.parents[0].type === "dict") {
+                    newValue.properties.push(addtionalProperties(newValue.parents[0]));
```

Comment:
This feels hacky to do that as a property. This is a metadata of the model. We may even ignore it entirely, as Python do not change the generated code based on additionalProperty information.